src/lambda_toolshed/papillon.cljc

```diff
@@ -43,146 +47,105 @@
   "Transition the context `ctx` to the candidate context value `candidate`.  This function
   works synchronously with value candidates -any async processing should be performed prior
   to invoking this function."
-  [ctx candidate-ctx tag]
+  [ctx tag candidate-ctx]
   (cond
     (-> candidate-ctx reduced?) (-> ctx
-                                    (transition (unreduced candidate-ctx) tag)
+                                    (transition tag (unreduced candidate-ctx))
                                     clear-queue)
     (-> candidate-ctx error?) (-> ctx
                                   (assoc ::error candidate-ctx)
+                                  (assoc ::stage :error)
```

Comment:
another possible candidate for combining the `assoc` calls.

Reply:
Yep.  On it.

Reply:
cool 👍